### PR TITLE
Implemented test case for JwtMiddleware #4418

### PR DESCRIPTION
**Pull request by nikzayn**
## Proposed changes
- Expected to add the test cases for JWT Middleware 

## Types of changes
- Added the test cases for JWT Middleware

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)

## Dependency
- (#4410)

## Special notes for your reviewer:
- Retest the changes

### CODE REVIEW

**Comment by Saranya-jena** [comment]
@nikzayn Can you please attach screenshots for the executed testcases and coverage?

**Comment by SarthakJain26** [comment]
@nikzayn please address the comments, so that we can merge this PR

**Comment by nikzayn** [comment]
> @nikzayn Can you please attach screenshots for the executed testcases and coverage?

Sorry, for the delay. I was actually out of town. I came back yesterday. I will update you on the following soon..!

**Comment by nikzayn** [comment]
> @nikzayn please address the comments, so that we can merge this PR

I will update for sure..!!

**Comment by nikzayn** [comment]
Here are the screenshots of test execution and coverage output.

Screenshot 2024-03-08 at 1 11 49 PM
Screenshot 2024-03-08 at 1 27 35 PM

File: chaoscenter/authentication/api/middleware/jwt_middleware_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type MockApplicationService struct{}
+
+// AddMember implements services.ApplicationService.
+func (m *MockApplicationService) AddMember(projectID string, member *entities.Member) error {
```
Comment (SarthakJain26) [inline chaoscenter/authentication/api/middleware/jwt_middleware_test.go:22]: @nikzayn you can remove all these generated functions that are unimplemented.

File: chaoscenter/authentication/api/middleware/jwt_middleware_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	// Subtest for Valid Token
+	t.Run("Valid Token", func(t *testing.T) {
+		wValid := createRequest("jwtstring")
+		assert.Equal(t, http.StatusOK, wValid.Code)
+	})
+
+	// Subtest for Missing Authorization Header
+	t.Run("Missing Authorization Header", func(t *testing.T) {
```
Comment (SarthakJain26) [inline chaoscenter/authentication/api/middleware/jwt_middleware_test.go:228]: I see there's 30% code coverage with this TC, please check if you can add more TCs for more code coverage

File: chaoscenter/authentication/api/middleware/coverage.html [hunk 3]
```diff
@@ -0,0 +1,142 @@
+
```
Comment (SarthakJain26) [inline chaoscenter/authentication/api/middleware/coverage.html:1]: @nikzayn you can remove both the coverage files, you don't need to commit those generated files.

**Review by SarthakJain26 — CHANGES_REQUESTED** [review]

**Comment by Saranya-jena** [comment]
@nikzayn are you still working on this PR?

**Comment by Saranya-jena** [comment]
closing due to inactivity

**Pull request closed without merging**